fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/IndexTypes.java

```diff
@@ -142,6 +142,11 @@
      */
     public static final String TEXT = "text";
 
+    /**
+     * An index on the tokens in a text field.
+     */
+    public static final String LUCENE = "lucene";
```

Comment:
Index types not defined in the core usually don't get named there, but rather in the .jar that adds them.

For instance, `fdb-record-layer-spatial/src/main/java/com/apple/foundationdb/record/spatial/geophile/GeophileIndexTypes`.